mlflow/tracking/_model_registry/client.py

```diff
@@ -191,7 +192,21 @@ def create_model_version(
         """
         tags = tags if tags else {}
         tags = [ModelVersionTag(key, str(value)) for key, value in tags.items()]
-        mv = self.store.create_model_version(name, source, run_id, tags, run_link, description)
+        try:
+            mv = self.store.create_model_version(
+                name,
+                source,
+                run_id,
+                tags,
+                run_link,
+                description,
+                local_model_path=local_model_path,
+            )
+        except TypeError:
```

Comment:
What if calling `self.store.create_model_version` succeeds, but `TypeError` is thrown in it?


```
def create_model_version(...):
    ...
    raise TypeError("something went wrong")
```

Reply:
Good question, we are assuming here that `self.store.create_model_version` will not perform other operations that could throw a `TypeError` after actually creating the model version, which I think is the case at least for all of the built-in MLflow stores. In the scenario you described it's possible we'd create two model versions. We could try to derisk that by checking the error message of the `TypeError`, but that feels a little brittle (e.g. if Python changes its error message format in a minor release, the logic would break)

Reply:
Can we inspect `self.store.create_model_version` using the `inpsect` module?

Reply:
or can we check the error message?

Reply:
@harupy yeah! I'll give that a try

Reply:
Seems to work; if I remove `local_model_path` support from the model registry filestore & try to register a model without the `inspect` check, it fails:

```
mlflow-dev ❯ python                                                                                                                                            [07/10/23 | 1:39:48]
Python 3.8.13 (default, Jun  6 2023, 21:17:02) 
[Clang 14.0.3 (clang-1403.0.22.14.1)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import mlflow
>>> 
>>> mlflow.register_model
<function register_model at 0x14cc54670>
>>> help(mlflow.register_model)

>>> mlflow.register_model("/tmp/blah")
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: register_model() missing 1 required positional argument: 'name'
>>> mlflow.register_model("/tmp/blah", name="Abcd")
Successfully registered model 'Abcd'.
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/sid.murching/mlflow/mlflow/tracking/_model_registry/fluent.py", line 73, in register_model
    return _register_model(
  File "/Users/sid.murching/mlflow/mlflow/tracking/_model_registry/fluent.py", line 108, in _register_model
    create_version_response = client._create_model_version(
  File "/Users/sid.murching/mlflow/mlflow/tracking/client.py", line 2570, in _create_model_version
    return self._get_registry_client().create_model_version(
  File "/Users/sid.murching/mlflow/mlflow/tracking/_model_registry/client.py", line 196, in create_model_version
    mv = self.store.create_model_version(
TypeError: create_model_version() got an unexpected keyword argument 'local_model_path'
```

But with the check it succeeds